web/src/engine/main/src/keyboardInterface.ts

```diff
@@ -28,16 +28,39 @@ export default class KeyboardInterface extends KeyboardInterfaceBase {
     this.stubAndKeyboardCache = cache;
   }
 
+  // Preserves a keyboard's ID, even if namespaced, via script tag tagging.
+  preserveID(Pk: any /** a `Keyboard`'s `scriptObject` entry */) {
```

Comment:
Where does the `<SCRIPT>` element get its id set? I don't really understand why this was added as part of esmodule changes?

Reply:
... it wasn't added.  It's being preserved.

https://github.com/keymanapp/keyman/blob/d2cacd58f30a885317919b07622ec406c165ad09/web/src/app/embed/kmwembedded.ts#L176-L197

For non-embedded contexts, it had a 'stub' implementation.

Set originally by:

https://github.com/keymanapp/keyman/blob/d2cacd58f30a885317919b07622ec406c165ad09/web/src/engine/main/keyboards/kmwkeyboards.ts#L713-L724

As `KeyboardInterface` is expected to exist by loading/parsing keyboards (hence it serving as a "harness"), this seemed the "right" place to put it.

The place where the ID is now set: 

https://github.com/keymanapp/keyman/blob/b4fee54336bcee583815e8e8d292550dfa8a210b/common/web/keyboard-processor/src/keyboards/loaders/domKeyboardLoader.ts#L31-L43

... right, I should add a guard that 'trueID' is defined.